plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -49,7 +54,7 @@
     /**
      * (acceskey, endpoint) -&gt; client
      */
-    private Map<Tuple<String, String>, AmazonS3Client> clients = new HashMap<>();
+    private Map<Tuple<String, Tuple<String, EncryptionMaterials>>, AmazonS3Client> clients = new HashMap<>();
```

Comment:
I don't understand why we need to add the `EncryptionMaterials` here. Does it mean that we want to have many client instances using the same account but with different encryption settings?